lib/modules.nix

```
@@ -1023,12 +1025,32 @@ let
       inherit priority content;
     };
 
-  mkOptionDefault = mkOverride 1500; # priority of option defaults
-  mkDefault = mkOverride 1000; # used in config sections of non-user modules to set a default
-  defaultOverridePriority = 100;
-  mkImageMediaOverride = mkOverride 60; # image media profiles can be derived by inclusion into host config, hence needing to override host config, but do allow user to mkForce
-  mkForce = mkOverride 50;
-  mkVMOverride = mkOverride 10; # used by ‘nixos-rebuild build-vm’
+  priorities = {
+    # Priority of options' default values
+    optionDefault = 1500;
+    # Used in config sections of non-user modules to set a default.
+    # (This is not the default priority assigned to values!)
+    configDefault = 1000;
+    # The priority that values get assigned by default around which other
+    # priorities scale. This is the true "default" priority.
+    baseline = 100;
+    # image media profiles can be derived by inclusion into host config, hence
+    # needing to override host config, but do allow user to mkForce
+    mediaOverride = 60;
+    # Forcefully override a value
+    force = 50;
+    # Used by ‘nixos-rebuild build-vm’
+    vmOverride = 10;
+  };
+  mkOptionDefault = mkOverride priorities.optionDefault;
+  mkConfigDefault = mkOverride priorities.configDefault;
+  mkImageMediaOverride = mkOverride priorities.mediaOverride;
+  mkForce = mkOverride priorities.force;
+  mkVMOverride = mkOverride priorities.vmOverride;
+
+  # Legacy aliases
+  mkDefault = mkConfigDefault;
+  defaultOverridePriority = priorities.baseline; # Deprecated! Please use priorities.baseline instead.
```

I don't think we're in a hurry, but let's plan the migration to start in July when the alternative is widely available.

```suggestion
  defaultOverridePriority = warnIf (oldestSupportedReleaseIsAtLeast 2505) "lib.modules.defaultOverridePriority is deprecated. Please use lib.modules.priorities.baseline instead." priorities.baseline;
```

---

Good idea, I'll mark both as deprecated as of next year and will check for usages inside of Nixpkgs.